services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events_user_services.py

```diff
@@ -122,11 +123,25 @@ async def progress_create_containers(
     users_repository = get_repository(app, UsersRepository)
     user_email = await users_repository.get_user_email(scheduler_data.user_id)
 
+    # Ask resource usage tracker for pricing plan
+    if scheduler_data.wallet_info:
+        resource_usage_api = ResourceUsageApi.get_from_state(app)
+        (
+            pricing_plan_id,
+            pricing_detail_id,
+        ) = await resource_usage_api.get_default_pricing_plan_and_pricing_detail_for_service(
+            scheduler_data.product_name, scheduler_data.key, scheduler_data.version
+        )
+
     metrics_params = CreateServiceMetricsAdditionalParams(
-        wallet_id=None,
-        wallet_name=None,
-        pricing_plan_id=None,
-        pricing_detail_id=None,
+        wallet_id=scheduler_data.wallet_info.wallet_id
+        if scheduler_data.wallet_info
+        else None,
+        wallet_name=scheduler_data.wallet_info.wallet_name
+        if scheduler_data.wallet_info
+        else None,
+        pricing_plan_id=pricing_plan_id if scheduler_data.wallet_info else None,
+        pricing_detail_id=pricing_detail_id if scheduler_data.wallet_info else None,
```

Comment:
please set these fields to None above the `if schduler_data.wallet_info` condition and then complete them inside, instead of using so many ifs